var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -163,28 +168,38 @@ def determine_b2_options(self, spec, options):
             threadingOpts.append('single')
         if not threadingOpts:
             raise RuntimeError("At least one of {singlethreaded, multithreaded} must be enabled")
-        
+
         options.extend([
             'toolset=%s' % self.determine_toolset(spec),
             'link=%s' % ','.join(linkTypes),
             '--layout=tagged'])
-        
+
         return threadingOpts
 
     def install(self, spec, prefix):
+        # On Darwin, Boost expects the Darwin libtool. However, one of the
+        # dependencies may have pulled in Spack's GNU libtool, and these two are
+        # not compatible. We thus create a symlink to Darwin's libtool and add
+        # it at the beginning of PATH.
+        if sys.platform == 'darwin':
```

Comment:
Using `package.yaml` to wrap Darwin's libtool (and others) prevents me from installing other versions of these packages. That's quite limiting. Yes, there should be a simple way to use an existing system package, but that's not limited to libtool or Darwin -- there's a recent discussion in this respect.

Of course, there's nothing wrong with allowing people to use such a `package.yaml`, or providing one for download, but one shouldn't be expected to use it on Darwin, since installing packages from source works fine as well.

My goal is to be able to provide an identical (as in: identical versions) of software packages across a broad range of systems. Wrapping e.g. Darwin's BLAS as you suggest means that certain other features that e.g. OpenBLAS provides would then not be available. I realize that other people have other goals, but this ideal -- being able to install e.g. OpenBLAS 0.2.16, and have it then "just work" -- should remain possible on Darwin in the default setup.

I'm against globally using Darwin's libtool since GNU libtool and Darwin libtool are different pieces of software. They should correspond to different packages (with different names). Maybe there should be a virtual package as an umbrella. Maybe the conflict that we're seeing here should be better handled by Spack (via introducing build-time dependencies that would avoid this conflict). But if a package wants to use GNU libtool on Darwin, this should remain possible.